### PR TITLE
Containerfix

### LINKED ISSUE #341: Sync bug with Fluidic Compressor in MP
Occured to me in MP. If one player changes the mode on the Fluidic Compressor while another had the GUI open, too, he won't see the update.

I actually have plans on how to fix that. But that would require to change the current two buttons to one button. I wanted to discuss this fix with you.

### PR DESCRIPTION
**Pull request by BigXplosion**
this should hopefully fix #341, but if I remember correctly I made the `redraw` function for this purpose

### CODE REVIEW

**Comment by AEnterprise** [comment]
should fix? could you maybe test that then?

**Comment by BigXplosion** [comment]
it doesn't crash and normal behavior isn't changed but I can't test on a server... especially as I need 2 people

**Comment by iTitus** [comment]
Start a server and two clients in your dev env, join localhost and then test that. It lags, but it works...

File: src/main/java/buildcraftAdditions/inventories/containers/ContainerFluidicCompressor.java [hunk 1]
```diff
@@ -78,6 +80,7 @@ public void updateProgressBar(int id, int value) {
 		switch (id) {
 			case 0:
 				inventory.fill = value != 0;
+				((GuiBase) ClientProxy.getOpenGui()).redraw();
```
Comment (iTitus) [inline src/main/java/buildcraftAdditions/inventories/containers/ContainerFluidicCompressor.java]: Please add an 'instanceof GuiBase' check here, you'll never know...

File: src/main/java/buildcraftAdditions/inventories/containers/ContainerItemSorter.java [hunk 2]
```diff
@@ -64,7 +66,9 @@ public void detectAndSendChanges() {
 	@SideOnly(Side.CLIENT)
 	public void updateProgressBar(int id, int value) {
 		super.updateProgressBar(id, value);
-		if (id >= 0 && id < colors.length)
+		if (id >= 0 && id < colors.length) {
 			inventory.colors[id] = (byte) (0xFF & value);
+			((GuiBase) ClientProxy.getOpenGui()).redraw();
```
Comment (iTitus) [inline src/main/java/buildcraftAdditions/inventories/containers/ContainerItemSorter.java]: Same here

File: src/main/java/buildcraftAdditions/inventories/containers/ContainerFluidicCompressor.java (hunk 1 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/inventories/containers/ContainerFluidicCompressor.java]: Also do BuildcraftAdditions.proxy here

File: src/main/java/buildcraftAdditions/proxy/ClientProxy.java [hunk 3]
```diff
@@ -102,4 +103,8 @@ public int addArmor(String name) {
 	public void addListeners() {
 		FMLCommonHandler.instance().bus().register(new KeyListener());
 	}
+
+	public static GuiScreen getOpenGui() {
```
Comment (iTitus) [inline src/main/java/buildcraftAdditions/proxy/ClientProxy.java]: PLease add that method to the Commonproxy too

Comment (BigXplosion) [inline src/main/java/buildcraftAdditions/proxy/ClientProxy.java]: GuiScreen doesn't exist on the server so not in the commonproxy. I just put this here as this was the best place I could thing of to place the method

Comment (iTitus) [inline src/main/java/buildcraftAdditions/proxy/ClientProxy.java]: Just return null in the CommonProxy and the currentScreen in the ClientProxy

**Pull request merged**